servers/rest-services/src/main/java/org/projectnessie/services/rest/RestV2TreeResource.java

```diff
@@ -107,7 +116,41 @@ private ContentService content() {
   @JsonView(Views.V2.class)
   @Override
   public ReferencesResponse getAllReferences(ReferencesParams params) {
-    return tree().getAllReferences(params.fetchOption(), params.filter());
+    Integer maxRecords = params.maxRecords();
+    return tree()
+        .getAllReferences(
+            params.fetchOption(),
+            params.filter(),
+            params.pageToken(),
+            new PagedResponseHandler<
+                ImmutableReferencesResponse.Builder, ReferencesResponse, Reference>() {
+              final int max = maxRecords != null ? Math.max(maxRecords, 0) : 0;
+
+              @Override
+              public ImmutableReferencesResponse.Builder newBuilder() {
+                return ReferencesResponse.builder();
+              }
+
+              @Override
+              public ReferencesResponse build(ImmutableReferencesResponse.Builder builder) {
+                return builder.build();
+              }
+
+              @Override
+              public boolean addEntry(
+                  ImmutableReferencesResponse.Builder builder, int cnt, Reference entry) {
+                if (max > 0 && cnt > max) {
```

Comment:
We will truncate this to some default server-side limit when paging is fully supported, right?

Reply:
The current behavior's unchanged - i.e. for references there's no default limit.